common/containers/container.go

```diff
@@ -0,0 +1,296 @@
+		return noAliases
+	}
+	return c.aliases
+}
+
+// candidatesWithAlias returns the resolved candidates from the container with any applicable
+// alias appended on the end of the list.
+func (c *Container) candidatesWithAlias(candidates []string, name string) []string {
+	// If an alias exists for the name, ensure it is searched last.
+	alias, found := c.aliasSet()[name]
+	if found {
+		return append(candidates, alias)
+	}
+	return candidates
+}
+
+// ContainerOption specifies a functional configuration option for a Container.
+//
+// Note, ContainerOption implementations must be able to handle nil container inputs.
+type ContainerOption func(*Container) (*Container, error)
+
+// Abbrevs configures a set of simple names as abbreviations for fully-qualified names.
+//
+// An alias can be useful when working with variables, functions, and especially types from
+// multiple namespaces:
+//
+//    // CEL object construction
+//    qual.pkg.version.ObjTypeName{
+//       field: alt.container.ver.FieldTypeName{value: ...}
+//    }
+//
+// Only one qualified path may be used as CEL container, so at least one of these references is
+// a long qualified name within an otherwise short CEL program. Using the following abbreviations,
+// the program becomes much simpler:
+//
+//    // CEL Go option
+//    Abbrevs("qual.pkg.version.ObjTypeName", "alt.container.ver.FieldTypeName")
+//    // Simplified Object construction
+//    ObjTypeName{field: FieldTypeName{value: ...}}
+//
+// There are a few rules for the qualified names and the simple abbreviations generated from them:
+// - Qualified names must be dot-delimited, e.g. `package.subpkg.name`.
+// - The last element in the qualified name is the simple name used as the abbreviation.
+// - Abbreviations names must not collide with each other.
+// - The abbreviation name must not collide with the root-level container name.
+//
+// Abbrevs are distinct from container-based references in the following important ways:
+// - Containers follow C++ namespace resolution rules with searches from the most qualified name
+//   to the least qualified name.
+// - Container references within the CEL program may be relative, and are resolved to fully
+//   qualified names at either type-check time or program plan time, whichever comes first.
+// - Abbreviations must resolve to a fully-qualified name.
+// - Resolved abbreviations do not participate in namespace resolution.
+// - Resolved abbreviations are searched after container names, including container names in the
+//   global scope.
+//
+// If there is ever a case where an identifier could be in both the container and in the alias,
+// the container wins as the container will continue to evolve over time and the program must be
+// forward compatible with changes in the container.
+func Abbrevs(qualifiedNames ...string) ContainerOption {
+	return func(c *Container) (*Container, error) {
+		for _, qn := range qualifiedNames {
+			alias := qn
```

Comment:
Not used - can initialize `alias` on line 200 below.

Reply:
Done